core/CMakeLists.txt

```diff
@@ -1,30 +1,58 @@
-message(STATUS "Building 'detray::core' component")
-
-add_library(detray_core INTERFACE)
-
-target_include_directories(detray_core
-  INTERFACE
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-    $<INSTALL_INTERFACE:include>
-)
-
-target_link_libraries(detray_core INTERFACE vecmem::core)
-
-install(
-  DIRECTORY include/core
-  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
 
-install(
-  DIRECTORY include/masks
-  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-
-install(
-  DIRECTORY include/utils
-  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-
-install(
-  DIRECTORY include/tools
-  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-
-add_library(detray::core ALIAS detray_core)
+# Let the user know what's happening.
+message(STATUS "Building 'detray::core' component")
 
+# Set up the library.
+detray_add_library( detray_core core
+   "include/detray/core/detector.hpp"
```

Comment:
So will add every new header file here?

Reply:
You don't have to. These libraries will work correctly, even if not every header file is listed rigorously in them.

At one point I plan to add code like the following to the `<foo>_add_library(...)` calls in all of our projects:

https://gitlab.cern.ch/atlas/atlasexternals/-/blob/master/Build/AtlasCMake/modules/AtlasInternals.cmake#L246-270

So that header and source files would show up in a "neat way" in IDEs like [Xcode](https://developer.apple.com/xcode/) and [Visual Studio](https://visualstudio.microsoft.com/). The reason that I didn't start with this yet, is that `INTERFACE` libraries generally show up terribly in IDEs. :frowning:

But at one point I do want to get to this. And if at that point all of the `<foo>_add_library(...)` calls will have the header files rigorously listed, I won't have to do a whole lot of tweaking in those PRs anymore. :wink:

Reply:
I admit I have no idea about this, but then I can look at them sorted by library/executable instead of file path?

Reply:
In IDEs you generally sort sources by "component". (library/executable/etc.) But unless you plan on using an actual IDE, this will not affect you too much.

See https://github.com/acts-project/vecmem/pull/143 for some more details on what this is about. :wink:

Reply:
Cool, I will!